CI: h5py-related failures in develop branch Azure CI

The Azure CI Windows jobs are failing on current `develop` branch/merges. The Azure logs tend to disappear after 10 days or so, so I'll summarize the types of failures:

- Python 3.7 32-bit: fails to `pip` install `h5py` version `3.0.0`
- Python 3.6, 3.7, 3.8 64-bit: 5 test suite failures in `TestH5MDReader`:

<details>

```
2020-10-30T20:10:03.6709718Z ================================== FAILURES ===================================
2020-10-30T20:10:03.6710719Z __________________________ TestH5MDReader.test_iter ___________________________
2020-10-30T20:10:03.6711428Z [gw1] win32 -- Python 3.8.6 C:\hostedtoolcache\windows\Python\3.8.6\x64\python.exe
2020-10-30T20:10:03.6711823Z 
2020-10-30T20:10:03.6712316Z self = <MDAnalysisTests.coordinates.test_h5md.TestH5MDReader object at 0x000001D67EDE0E50>
2020-10-30T20:10:03.6712867Z ref = <MDAnalysisTests.coordinates.test_h5md.H5MDReference object at 0x000001D67EDE0A90>
2020-10-30T20:10:03.6713508Z reader = <H5MDReader D:\a\1\s\testsuite\MDAnalysisTests\data\coordinates\test.h5md with 5 frames of 5 atoms>
2020-10-30T20:10:03.6713923Z 
2020-10-30T20:10:03.6714344Z     def test_iter(self, ref, reader):
2020-10-30T20:10:03.6714819Z >       for i, ts in enumerate(reader):
2020-10-30T20:10:03.6715149Z 
2020-10-30T20:10:03.6715554Z MDAnalysisTests\coordinates\base.py:292: 
2020-10-30T20:10:03.6716033Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.6716783Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1474: in __next__
2020-10-30T20:10:03.6717443Z     return self.next()
2020-10-30T20:10:03.6718134Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1460: in next
2020-10-30T20:10:03.6719628Z     ts = self._read_next_timestep()
2020-10-30T20:10:03.6720540Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:692: in _read_next_timestep
2020-10-30T20:10:03.6721412Z     return self._read_frame(self._frame + 1)
2020-10-30T20:10:03.6722182Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:598: in _read_frame
2020-10-30T20:10:03.6723050Z     _ = self._particle_group[name]['step'][frame]
2020-10-30T20:10:03.6723695Z h5py\_objects.pyx:54: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.6724175Z     ???
2020-10-30T20:10:03.6724701Z h5py\_objects.pyx:55: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.6725111Z     ???
2020-10-30T20:10:03.6725769Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\h5py\_hl\dataset.py:704: in __getitem__
2020-10-30T20:10:03.6726291Z     return self._fast_reader.read(args)
2020-10-30T20:10:03.6726786Z h5py\_selector.pyx:344: in h5py._selector.Reader.read
2020-10-30T20:10:03.6727215Z     ???
2020-10-30T20:10:03.6727621Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.6728037Z 
2020-10-30T20:10:03.6728329Z >   ???
2020-10-30T20:10:03.6728754Z E   IndexError: Index (5) out of range for (0-4)
2020-10-30T20:10:03.6729070Z 
2020-10-30T20:10:03.6729470Z h5py\_selector.pyx:148: IndexError
2020-10-30T20:10:03.6730006Z __________________ TestH5MDReader.test_transformations_2iter __________________
2020-10-30T20:10:03.6730575Z [gw1] win32 -- Python 3.8.6 C:\hostedtoolcache\windows\Python\3.8.6\x64\python.exe
2020-10-30T20:10:03.6731012Z 
2020-10-30T20:10:03.6731421Z self = <MDAnalysisTests.coordinates.test_h5md.TestH5MDReader object at 0x000001D6026A4DC0>
2020-10-30T20:10:03.6732027Z ref = <MDAnalysisTests.coordinates.test_h5md.H5MDReference object at 0x000001D6026A4AF0>
2020-10-30T20:10:03.6732660Z transformed = <H5MDReader D:\a\1\s\testsuite\MDAnalysisTests\data\coordinates\test.h5md with 5 frames of 5 atoms>
2020-10-30T20:10:03.6733087Z 
2020-10-30T20:10:03.6733532Z     def test_transformations_2iter(self, ref, transformed):
2020-10-30T20:10:03.6733996Z         # Are the transformations applied and
2020-10-30T20:10:03.6734466Z         # are the coordinates "overtransformed"?
2020-10-30T20:10:03.6735028Z         v1 = np.float32((1,1,1))
2020-10-30T20:10:03.6735476Z         v2 = np.float32((0,0,0.33))
2020-10-30T20:10:03.6735890Z         idealcoords=[]
2020-10-30T20:10:03.6736400Z >       for i, ts in enumerate(transformed):
2020-10-30T20:10:03.6736792Z 
2020-10-30T20:10:03.6737148Z MDAnalysisTests\coordinates\base.py:372: 
2020-10-30T20:10:03.6737677Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.6738408Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1474: in __next__
2020-10-30T20:10:03.6739017Z     return self.next()
2020-10-30T20:10:03.6739702Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1460: in next
2020-10-30T20:10:03.6740435Z     ts = self._read_next_timestep()
2020-10-30T20:10:03.6741084Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:692: in _read_next_timestep
2020-10-30T20:10:03.6741790Z     return self._read_frame(self._frame + 1)
2020-10-30T20:10:03.6742465Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:598: in _read_frame
2020-10-30T20:10:03.6743106Z     _ = self._particle_group[name]['step'][frame]
2020-10-30T20:10:03.6743615Z h5py\_objects.pyx:54: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.6744486Z     ???
2020-10-30T20:10:03.6744891Z h5py\_objects.pyx:55: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.6745336Z     ???
2020-10-30T20:10:03.6746078Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\h5py\_hl\dataset.py:704: in __getitem__
2020-10-30T20:10:03.6746738Z     return self._fast_reader.read(args)
2020-10-30T20:10:03.6747276Z h5py\_selector.pyx:344: in h5py._selector.Reader.read
2020-10-30T20:10:03.6747708Z     ???
2020-10-30T20:10:03.6748238Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.6748636Z 
2020-10-30T20:10:03.6749097Z >   ???
2020-10-30T20:10:03.6749565Z E   IndexError: Index (5) out of range for (0-4)
2020-10-30T20:10:03.6749984Z 
2020-10-30T20:10:03.6750457Z h5py\_selector.pyx:148: IndexError
2020-10-30T20:10:03.6750983Z __________________ TestH5MDReader.test_transformations_copy ___________________
2020-10-30T20:10:03.6751645Z [gw1] win32 -- Python 3.8.6 C:\hostedtoolcache\windows\Python\3.8.6\x64\python.exe
2020-10-30T20:10:03.6752078Z 
2020-10-30T20:10:03.6752843Z self = <MDAnalysisTests.coordinates.test_h5md.TestH5MDReader object at 0x000001D6084F8FA0>
2020-10-30T20:10:03.6753846Z ref = <MDAnalysisTests.coordinates.test_h5md.H5MDReference object at 0x000001D6084F88E0>
2020-10-30T20:10:03.6754581Z transformed = <H5MDReader D:\a\1\s\testsuite\MDAnalysisTests\data\coordinates\test.h5md with 5 frames of 5 atoms>
2020-10-30T20:10:03.6755116Z 
2020-10-30T20:10:03.6755561Z     def test_transformations_copy(self,ref,transformed):
2020-10-30T20:10:03.6756199Z         # this test checks if transformations are carried over a copy and if the
2020-10-30T20:10:03.6756741Z         # coordinates of the copy are equal to the original's
2020-10-30T20:10:03.6757491Z         v1 = np.float32((1,1,1))
2020-10-30T20:10:03.6758686Z         v2 = np.float32((0,0,0.33))
2020-10-30T20:10:03.6759454Z         new = transformed.copy()
2020-10-30T20:10:03.6760183Z         assert_equal(transformed.transformations, new.transformations,
2020-10-30T20:10:03.6760817Z                      "transformations are not equal")
2020-10-30T20:10:03.6761322Z >       for i, ts in enumerate(new):
2020-10-30T20:10:03.6761745Z 
2020-10-30T20:10:03.6762151Z MDAnalysisTests\coordinates\base.py:425: 
2020-10-30T20:10:03.6762753Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.6763737Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1474: in __next__
2020-10-30T20:10:03.6764693Z     return self.next()
2020-10-30T20:10:03.6765443Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1460: in next
2020-10-30T20:10:03.6766144Z     ts = self._read_next_timestep()
2020-10-30T20:10:03.6766836Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:692: in _read_next_timestep
2020-10-30T20:10:03.6767592Z     return self._read_frame(self._frame + 1)
2020-10-30T20:10:03.6768256Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:598: in _read_frame
2020-10-30T20:10:03.6768999Z     _ = self._particle_group[name]['step'][frame]
2020-10-30T20:10:03.6769545Z h5py\_objects.pyx:54: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.6770037Z     ???
2020-10-30T20:10:03.6770557Z h5py\_objects.pyx:55: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.6771023Z     ???
2020-10-30T20:10:03.6771538Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\h5py\_hl\dataset.py:704: in __getitem__
2020-10-30T20:10:03.6772156Z     return self._fast_reader.read(args)
2020-10-30T20:10:03.6772658Z h5py\_selector.pyx:344: in h5py._selector.Reader.read
2020-10-30T20:10:03.6773065Z     ???
2020-10-30T20:10:03.6774067Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.6774461Z 
2020-10-30T20:10:03.6774846Z >   ???
2020-10-30T20:10:03.6775225Z E   IndexError: Index (5) out of range for (0-4)
2020-10-30T20:10:03.6775624Z 
2020-10-30T20:10:03.6775981Z h5py\_selector.pyx:148: IndexError
2020-10-30T20:10:03.6776524Z ____________________ TestH5MDReader.test_iter_as_aux_highf ____________________
2020-10-30T20:10:03.6777136Z [gw1] win32 -- Python 3.8.6 C:\hostedtoolcache\windows\Python\3.8.6\x64\python.exe
2020-10-30T20:10:03.6777614Z 
2020-10-30T20:10:03.6778116Z self = <MDAnalysisTests.coordinates.test_h5md.TestH5MDReader object at 0x000001D608B03430>
2020-10-30T20:10:03.6778700Z ref = <MDAnalysisTests.coordinates.test_h5md.H5MDReference object at 0x000001D608B03A60>
2020-10-30T20:10:03.6779379Z reader = <H5MDReader D:\a\1\s\testsuite\MDAnalysisTests\data\coordinates\test.h5md with 5 frames of 5 atoms>
2020-10-30T20:10:03.6779808Z 
2020-10-30T20:10:03.6780369Z     def test_iter_as_aux_highf(self, ref, reader):
2020-10-30T20:10:03.6780940Z         # auxiliary has a higher frequency, so iter_as_aux should behave the
2020-10-30T20:10:03.6781451Z         # same as regular iteration over the trjectory
2020-10-30T20:10:03.6782016Z >       for i, ts in enumerate(reader.iter_as_aux('highf')):
2020-10-30T20:10:03.6782387Z 
2020-10-30T20:10:03.6782828Z MDAnalysisTests\coordinates\base.py:494: 
2020-10-30T20:10:03.6783369Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.6784118Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1873: in iter_as_aux
2020-10-30T20:10:03.6784836Z     yield self.next_as_aux(auxname)
2020-10-30T20:10:03.6785538Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1856: in next_as_aux
2020-10-30T20:10:03.6786183Z     ts = self.next()
2020-10-30T20:10:03.6786880Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1460: in next
2020-10-30T20:10:03.6787579Z     ts = self._read_next_timestep()
2020-10-30T20:10:03.6788260Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:692: in _read_next_timestep
2020-10-30T20:10:03.7250502Z     return self._read_frame(self._frame + 1)
2020-10-30T20:10:03.7252154Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:598: in _read_frame
2020-10-30T20:10:03.7253000Z     _ = self._particle_group[name]['step'][frame]
2020-10-30T20:10:03.7253443Z h5py\_objects.pyx:54: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.7253785Z     ???
2020-10-30T20:10:03.7254068Z h5py\_objects.pyx:55: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.7254404Z     ???
2020-10-30T20:10:03.7254814Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\h5py\_hl\dataset.py:704: in __getitem__
2020-10-30T20:10:03.7255238Z     return self._fast_reader.read(args)
2020-10-30T20:10:03.7255607Z h5py\_selector.pyx:344: in h5py._selector.Reader.read
2020-10-30T20:10:03.7255917Z     ???
2020-10-30T20:10:03.7256218Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.7256530Z 
2020-10-30T20:10:03.7256720Z >   ???
2020-10-30T20:10:03.7257036Z E   IndexError: Index (5) out of range for (0-4)
2020-10-30T20:10:03.7257266Z 
2020-10-30T20:10:03.7257547Z h5py\_selector.pyx:148: IndexError
2020-10-30T20:10:03.7257900Z __________________ TestH5MDReader.test_transformations_iter ___________________
2020-10-30T20:10:03.7258370Z [gw0] win32 -- Python 3.8.6 C:\hostedtoolcache\windows\Python\3.8.6\x64\python.exe
2020-10-30T20:10:03.7258649Z 
2020-10-30T20:10:03.7259005Z self = <MDAnalysisTests.coordinates.test_h5md.TestH5MDReader object at 0x0000027887092370>
2020-10-30T20:10:03.7260337Z ref = <MDAnalysisTests.coordinates.test_h5md.H5MDReference object at 0x0000027887092E80>
2020-10-30T20:10:03.7260842Z transformed = <H5MDReader D:\a\1\s\testsuite\MDAnalysisTests\data\coordinates\test.h5md with 5 frames of 5 atoms>
2020-10-30T20:10:03.7261224Z 
2020-10-30T20:10:03.7261512Z     def test_transformations_iter(self, ref, transformed):
2020-10-30T20:10:03.7261952Z         # test one iteration and see if the transformations are applied
2020-10-30T20:10:03.7262326Z         v1 = np.float32((1,1,1))
2020-10-30T20:10:03.7262600Z         v2 = np.float32((0,0,0.33))
2020-10-30T20:10:03.7262988Z >       for i, ts in enumerate(transformed):
2020-10-30T20:10:03.7263217Z 
2020-10-30T20:10:03.7263517Z MDAnalysisTests\coordinates\base.py:362: 
2020-10-30T20:10:03.7264086Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.7264730Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1474: in __next__
2020-10-30T20:10:03.7265330Z     return self.next()
2020-10-30T20:10:03.7265917Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\base.py:1460: in next
2020-10-30T20:10:03.7266512Z     ts = self._read_next_timestep()
2020-10-30T20:10:03.7267087Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:692: in _read_next_timestep
2020-10-30T20:10:03.7267721Z     return self._read_frame(self._frame + 1)
2020-10-30T20:10:03.7268355Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\mdanalysis-2.0.0.dev0-py3.8-win-amd64.egg\MDAnalysis\coordinates\H5MD.py:598: in _read_frame
2020-10-30T20:10:03.7268970Z     _ = self._particle_group[name]['step'][frame]
2020-10-30T20:10:03.7269362Z h5py\_objects.pyx:54: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.7269736Z     ???
2020-10-30T20:10:03.7270127Z h5py\_objects.pyx:55: in h5py._objects.with_phil.wrapper
2020-10-30T20:10:03.7270454Z     ???
2020-10-30T20:10:03.7270915Z C:\hostedtoolcache\windows\Python\3.8.6\x64\lib\site-packages\h5py\_hl\dataset.py:704: in __getitem__
2020-10-30T20:10:03.7271415Z     return self._fast_reader.read(args)
2020-10-30T20:10:03.7271786Z h5py\_selector.pyx:344: in h5py._selector.Reader.read
2020-10-30T20:10:03.7272151Z     ???
2020-10-30T20:10:03.7272531Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-10-30T20:10:03.7273110Z 
2020-10-30T20:10:03.7273584Z >   ???
2020-10-30T20:10:03.7274128Z E   IndexError: Index (5) out of range for (0-4)
2020-10-30T20:10:03.7274516Z 
2020-10-30T20:10:03.7274754Z h5py\_selector.pyx:148: IndexError
2020-10-30T20:10:03.7275019Z 
```

</details>

Failures started on October 30th, which was the exact release date for `h5py` `3.0.0`. So, I may try to fix this with a temporary pin to the previous stable release version `2.10.0`. There's already a `3.1.0`, but maybe I'll try to be conservative for those windows jobs for now...

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -54,10 +54,11 @@ jobs:
       pytest-xdist
       scikit-learn
       scipy
-      h5py
+      h5py==2.10.0
       tqdm
     displayName: 'Install dependencies'
   # TODO: recent rdkit is not on PyPI
+  # NOTE: h5py pinned because of gh-3019
   - script: >-
       python -m pip install
       biopython

